File: Source/AlphaTab/Rendering/Glyphs/DiamondNoteHeadGlyph.cs
using AlphaTab.Model;

namespace AlphaTab.Rendering.Glyphs
{
    internal class DiamondNoteHeadGlyph : MusicFontGlyph
    {
        private readonly bool _isGrace;

        public DiamondNoteHeadGlyph(float x, float y, Duration duration, bool isGrace)
            : base(x, y, isGrace ? NoteHeadGlyph.GraceScale : 1, GetSymbol(duration))
        {
            _isGrace = isGrace;
        }

        private static MusicFontSymbol GetSymbol(Duration duration)
        {
            switch (duration)
            {
                case Duration.QuadrupleWhole:
                case Duration.DoubleWhole:
                case Duration.Whole:
                case Duration.Half:
                    return MusicFontSymbol.NoteHarmonicWhole;
                default:
                    return MusicFontSymbol.NoteHarmonic;
            }
        }

        public override void DoLayout()
        {
            Width = 9 * (_isGrace ? NoteHeadGlyph.GraceScale : 1) * Scale;
        }
    }
}

File: src/rendering/glyphs/DiamondNoteHeadGlyph.ts
import { Duration } from '@src/model/Duration';
import { MusicFontGlyph } from '@src/rendering/glyphs/MusicFontGlyph';
import { MusicFontSymbol } from '@src/model/MusicFontSymbol';
import { NoteHeadGlyph } from '@src/rendering/glyphs/NoteHeadGlyph';

export class DiamondNoteHeadGlyph extends MusicFontGlyph {
    private _isGrace: boolean;

    public constructor(x: number, y: number, duration: Duration, isGrace: boolean) {
        super(x, y, isGrace ? NoteHeadGlyph.GraceScale : 1, DiamondNoteHeadGlyph.getSymbol(duration));
        this._isGrace = isGrace;
    }

    private static getSymbol(duration: Duration): MusicFontSymbol {
        switch (duration) {
            case Duration.QuadrupleWhole:
            case Duration.DoubleWhole:
            case Duration.Whole:
            case Duration.Half:
                return MusicFontSymbol.NoteheadDiamondWhiteWide;
            default:
                return MusicFontSymbol.NoteheadDiamondBlackWide;
        }
    }

    public doLayout(): void {
        this.width = 9 * (this._isGrace ? NoteHeadGlyph.GraceScale : 1) * this.scale;
        this.height = NoteHeadGlyph.NoteHeadHeight * this.scale;
    }
}
